searx/engines/core.py

```diff
@@ -23,92 +23,104 @@
 
 api_key = 'unset'
 
-base_url = 'https://core.ac.uk:443/api-v2/search/'
-search_string = '{query}?page={page}&pageSize={nb_per_page}&apiKey={apikey}'
-
+base_url = 'https://api.core.ac.uk/v3/search/works/'
 
 def request(query, params):
-
     if api_key == 'unset':
         raise SearxEngineAPIException('missing CORE API key')
 
-    search_path = search_string.format(
-        query=urlencode({'q': query}),
-        nb_per_page=nb_per_page,
-        page=params['pageno'],
-        apikey=api_key,
-    )
-    params['url'] = base_url + search_path
+    # API v3 uses different parameters
+    search_params = {
+        'q': query,
+        'offset': (params['pageno'] - 1) * nb_per_page,
+        'limit': nb_per_page,
+        'sort': 'relevance'
+    }
 
-    return params
+    params['url'] = base_url + '?' + urlencode(search_params)
+    params['headers'] = {'Authorization': f'Bearer {api_key}'}
 
+    return params
 
 def response(resp):
     results = []
     json_data = resp.json()
 
-    for result in json_data['data']:
-        source = result['_source']
+    for result in json_data.get('results', []):
+        # Get title
+        title = result.get('title')
+        if not title:
```

Comment:
You should probably either use the `title` variable at the bottom when creating the `result = {...}` or do a `if not result.get('title')` here, otherwise creating this variable for one time use is redundant.